docs/guides/big-data.md

```diff
@@ -0,0 +1,133 @@
+1. Compress results written to disk to save space.
+1. Use a [task runner](/concepts/task-runners/) for parallelizable operations to save time.
+
+### Remove task introspection
+
+When a task is called from a flow, each argument is introspected by Prefect, by default.
+To speed up your flow runs, you can disable this behavior for a task by wrapping the argument using [`quote`](https://docs.prefect.io/latest/api-ref/prefect/utilities/annotations/#prefect.utilities.annotations.quote).
+
+To demonstrate, let's use a basic example that extracts and transforms some New York taxi data.
+
+```python hl_lines="2 26" title="et_quote.py"
+from prefect import task, flow
+from prefect.utilities.annotations import quote
+import pandas as pd
+
+
+@task
+def extract(url: str):
+    """Extract data"""
+    df_raw = pd.read_parquet(url)
+    print(df_raw.info())
+    return df_raw
+
+
+@task
+def transform(df: pd.DataFrame):
+    """Basic transformation"""
+    df["tip_fraction"] = df["tip_amount"] / df["total_amount"]
+    print(df.info())
+    return df
+
+
+@flow(log_prints=True)
+def et(url: str):
+    """ET pipeline"""
+    df_raw = extract(url)
+    df = transform(quote(df_raw))
+
+
+if __name__ == "__main__":
+    url = "https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2023-09.parquet"
+    et(url)
+```
+
+Introspection can take significant time when the object being passed is a large collection, such as dictionary or DataFrame, where each element needs to be visited.
+Note that using `quote` improves performance at the expense of disabling task dependency tracking for the wrapped object.
+
+### Write task results to cloud storage
+
+By default, the results of task runs are stored in memory in your execution environment.
+This behavior helps make flow runs fast for small data, but can be problematic for large data.
+You can save memory by writing results to disk.
+In production, you'll generally want to write results to a cloud provider storage such as AWS S3.
+Prefect lets you to use a storage block from a Prefect cloud integration library such as [prefect-aws](https://prefecthq.github.io/prefect-aws/) to save your configuration information.
+Learn more about blocks [here](/concepts/blocks/).
+
+Install the relevant library, register the block with the server, and create your storage block.
+Then you can reference the block in your flow like this:
+
+```python
+...
+from prefect_aws.s3 import S3Bucket
+
+my_s3_block = S3Bucket.load("MY_BLOCK_NAME")
+
+...
+@task(result_storage=my_s3_block)
+
+```
+
+Now the result of the task will be written to S3, rather than stored in memory.
+
+### Save data to disk within a flow
+
+To save memory and time with big data, you don't need to pass results between tasks at all.
+Instead, you can write and read data to disk within a flow.
+Prefect has integration libraries for each of the major cloud providers.
+Each library contains blocks with methods that make it convenient to read and write data.
```

Comment:
Why do we mention blocks here? I thought the section above is about persisting data to object storage. This section is about the local disk, right?

Reply:
The idea I'm hoping to get across here is that you can just read and write data outside of results - just read in your data from (e.g.) S3 in your code using a block method, transform it if needed, write it back out. The moving data guide shows how to do that.

The section above showed how to persist data with results, so just another way to that, but not as useful in some cases e.g. you want to work with multiple files.

If you have ideas for better explaining that, I'm all ears. :)

Reply:
Ah I see, let me take a stab at that.